dev_cmds/integration-tests/single-package-json-example.ts

```diff
@@ -0,0 +1,71 @@
+import { execPiped, execToString } from "devcmd";
+import { red } from "kleur/colors";
+import { DOCKER_COMMAND } from "../utils/commands";
+import { NpmPackResult } from "../utils/npm-utils";
+import { singlePackageJsonExampleDir } from "../utils/paths";
+import { TestGroup, TestFunction, installDevcmdCliGlobally, LOCAL_REGISTRY_URL } from "./integration-test-harness";
+
+export function createSinglePackageJsonExampleTestGroup(devcmdCliInfo: NpmPackResult): TestGroup {
+  const setup: TestFunction = async (containerName: string) => {
+    await installDevcmdCliGlobally(containerName, devcmdCliInfo);
+
+    await execPiped({
+      command: DOCKER_COMMAND,
+      args: ["exec", containerName, "sh", "-c", "mkdir /tmp/devcmd_test"],
+    });
+
+    await execPiped({
+      command: DOCKER_COMMAND,
+      args: ["cp", singlePackageJsonExampleDir, `${containerName}:/tmp/devcmd_test`],
+    });
+
+    await execPiped({
+      command: DOCKER_COMMAND,
+      args: ["exec", "--user", "root", containerName, "chown", "-R", "verdaccio", "/tmp/devcmd_test"],
+    });
+
+    await execPiped({
+      command: DOCKER_COMMAND,
+      args: [
+        "exec",
+        containerName,
+        "sh",
+        "-c",
+        ["cd /tmp/devcmd_test/single-package-json", `npm --registry ${LOCAL_REGISTRY_URL} install`].join(" && "),
+      ],
+    });
+
+    return "success";
+  };
+
+  const runExampleCmd: TestFunction = async (containerName: string) => {
+    const exampleCmdCommandLine = `npx devcmd example_cmd`;
+    const { stdout, stderr } = await execToString({
+      command: DOCKER_COMMAND,
```

Comment:
I was wondering if we could abstract from the fact that tests are executed in docker in the integration test implementations (and just run the test implementations within the docker) - but then we probably can't use dev_cmd to check the test outcome.

Reply:
I like this thought, but I agree that it would probably require some restructuring of how the tests are run and evaluated. The test harness should probably still be usable for that, so let's discuss this as a separate change :)